### PR TITLE
IP-799: Kudobot Error Handling

### PR DESCRIPTION
**Pull request by evaninja**
**Details**: Prevents Kudobot crashes by surrounding in catch all logic (for now, useful in debugging), + error message handling.

Recommend whitespace-insensitive viewing for code change: https://github.com/Clever/node-slack-client/pull/5/files?w=1

**Testing**: Deployed in staging, checked logs

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -22,6 +22,6 @@ publish:
     - $(git rev-parse --short HEAD)
     username: $$docker_username
     when:
-      branch: master
+      branch: IP-799-kudobot-go
```
Comment (potluck) [inline .drone.yml]: don't forget to revert :)

**Comment by potluck** [comment]
> Recommend whitespace-insensitive viewing for code change: https://github.com/Clever/node-slack-client/pull/5/files?w=1

Thank you for that rec!

File: kudobot.coffee [hunk 2]
```diff
@@ -72,65 +72,70 @@ slack.on 'open', ->
...
-    """
+      typeError = if type isnt 'message' then "unexpected type #{type}." else null
+      textError = if not text? then 'text was undefined.' else null
+      undefChannelError = if not channel? then 'channel was undefined.' else null
+      channelError = if channel?.is_channel then 'post in public channel' else null
+      kudobotCase = if userName is "@kudobot" then 'Kudobot posted the message' else null
+
+      errors = [typeError, textError, undefChannelError, channelError, kudobotCase].filter((element) -> element isnt null).join ' '
+
+      console.log """
```
Comment (potluck) [inline kudobot.coffee]: Do we want to consider a better method than `console.log` for this error-surfacing? (here and below)

**Comment by potluck** [comment]
LGTM

File: kudobot.coffee (hunk 2 above)
Comment (evaninja) [inline kudobot.coffee]: Nice catch. Using `console.error` instead in 35143d8 :)

Comment (potluck) [inline kudobot.coffee]: 🚢

**Pull request merged**